src/Agent.Plugins/GitCliManager.cs

```diff
@@ -175,23 +175,34 @@ public async Task<int> GitFetch(AgentTaskPluginExecutionContext context, string
             string progress = reducedOutput ? string.Empty : "--progress";
 
             // default options for git fetch.
-            string options = StringUtil.Format($"{forceTag} --tags --prune {progress} --no-recurse-submodules {remoteName} {string.Join(" ", refSpec)}");
+            string fetchString = $"--tags --prune --progress --no-recurse-submodules {remoteName} {string.Join(" ", refSpec)}";
 
+            // insert prune-tags if knob is false to sync tags with the remote
+            if(!AgentKnobs.DisableFetchPruneTags.GetValue(context).AsBoolean()){
+                string modifiedFetchString = fetchString.Insert(fetchString.IndexOf("--progress"),"--prune-tags ");
+                fetchString = modifiedFetchString;
+            }
+            
             // If shallow fetch add --depth arg
             // If the local repository is shallowed but there is no fetch depth provide for this build,
             // add --unshallow to convert the shallow repository to a complete repository
             if (fetchDepth > 0)
-            {
-                options = StringUtil.Format($"{forceTag} --tags --prune {progress} --no-recurse-submodules --depth={fetchDepth} {remoteName} {string.Join(" ", refSpec)}");
+            {   
+                string modifiedFetchString = fetchString.Insert(fetchString.IndexOf($"{remoteName}"), $"--depth={fetchDepth} ");
+                fetchString = modifiedFetchString; 
             }
             else
             {
                 if (File.Exists(Path.Combine(repositoryPath, ".git", "shallow")))
                 {
-                    options = StringUtil.Format($"{forceTag} --tags --prune {progress} --no-recurse-submodules --unshallow {remoteName} {string.Join(" ", refSpec)}");
+                    
+                string modifiedFetchString = fetchString.Insert(fetchString.IndexOf($"{remoteName}"), "--unshallow ");
+                fetchString = modifiedFetchString;
                 }
             }
 
+            //define options for fetch
+            string options = StringUtil.Format(fetchString);
```

Comment:
Does this `StringUtil.Format` call do anything?

Reply:
Yes, formatting

Reply:
This method simply uses String.Format inside, which does not do anything with a string as far as I remember without arguments - so this looks redundant.
String interpolation does not require String.Format.

Reply:
Sure

Reply:
So why do we need the StringUtil.Format?